addons/delivery/models/sale_order.py

```diff
@@ -162,6 +162,9 @@ class SaleOrderLine(models.Model):
     product_qty = fields.Float(compute='_compute_product_qty', string='Product Qty', digits='Product Unit of Measure')
     recompute_delivery_price = fields.Boolean(related='order_id.recompute_delivery_price')
 
+    def _is_eligible_for_total(self):
+        return not self.is_delivery or super()._is_eligible_for_total()
```

Comment:
I think the logic should be inverted because you will return `True` for any non-delivery line, independently of the super value.